src/app/cli/src/init/test_persistent_frontier.ml

```diff
@@ -0,0 +1,76 @@
+open Persistent_frontier
+open Core
+
+let deserialize_root_hash ~logger ~db =
+  [%log info] "Querying root hash from database and attempt to deserialize it" ;
+  match Database.get_root_hash db with
+  | Ok hash ->
+      [%log info] "Got hash %s" (Pasta_bindings.Fp.to_string hash)
+  | Error _ ->
+      [%log error] "No root hash found"
+
+let root_hash_deserialization_limit_ms = 2000.0
+
+let main ~frontier_db_path ~no_root_compatible ~num_of_samples () =
+  let logger = Logger.create () in
+  [%log info] "Current DB path: %s" frontier_db_path ;
+  [%log info] "Loading database" ;
+  let db = Database.create ~logger ~directory:frontier_db_path in
+  if not no_root_compatible then
+    (*
+      To maintain compatibility, run a deserialization round first,
+      to ensure we have `root_hash` and `root_common` inside the DB
+      *)
+    deserialize_root_hash ~logger ~db ;
+  assert (Database.is_root_replaced_by_common_and_hash db) ;
+  let rec sample cnt =
+    match cnt with
+    | 0 ->
+        []
+    | _ ->
+        let start_time = Time_ns.now () in
```

Comment:
May I suggest that we use `Core_bench` library instead of measuring manually?

Let's set execution quota to 3 minutes?